manifests/key_pair.pp

```diff
@@ -0,0 +1,29 @@
+define certs::key_pair (
+  $source_dir,
+  $key_file,
+  $cert_file,
+  $key_owner   = undef,
+  $key_group   = undef,
+  $key_mode    = undef,
+  $cert_owner  = undef,
+  $cert_group  = undef,
+  $cert_mode   = undef,
+) {
+
+  file { $key_file:
+    ensure  => file,
+    source  => "${source_dir}/${title}.key",
+    owner   => $key_owner,
+    group   => $key_group,
+    mode    => $key_mode,
+  }
+
+  certificate_file { $cert_file:
```

Comment:
I added this new type to show how we could go with a provider for certificate file (and then one for private key) to provide and abstraction and additional validations. We should ask ourselves if this abstraction adds enough value on top of the File resource to be worth the additional layer.